### PR TITLE
Clarify startupProbe e2e tests

### PR DESCRIPTION
**Pull request by matthyx**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Make startupProbe e2e clearer to understand.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: test/e2e_node/startup_probe_test.go [hunk 1]
```diff
@@ -87,15 +116,15 @@ var _ = framework.KubeDescribe("StartupProbe [Serial] [Disruptive] [NodeAlphaFea
 
 		/*
 			Release : v1.16
-			Testname: Pod liveness probe, using local file, delayed by startup probe
-			Description: A Pod is created with liveness probe that uses ‘exec’ command to cat the non-existent /tmp/health file. Liveness probe MUST fail after startup probe expires. The Pod MUST now be killed and restarted incrementing restart count to 1.
+			Testname: Pod liveness probe delayed (short) by startup probe
+			Description: A Pod is created with failing liveness and startup probes. Liveness probe MUST fail after startup probe expires. The Pod MUST now be killed and restarted incrementing restart count to 1.
 		*/
-		framework.ConformanceIt("should be restarted with a exec \"cat /tmp/health\" because startup probe does not delay it long enough [NodeConformance]", func() {
+		framework.ConformanceIt("should be restarted by liveness probe because startup probe does not delay it long enough [NodeConformance]", func() {
```
Comment (odinuge) [inline test/e2e_node/startup_probe_test.go]: Shouldn't this container restart because startup probe fails?

**Review by odinuge — APPROVED** [review]
Thanks for clearing up the tests! This make much more sense now! :smile: 

/lgtm

File: test/e2e_node/startup_probe_test.go (hunk 1 above)
Comment (matthyx) [reply test/e2e_node/startup_probe_test.go]: Yup, this test doesn't make sense...

**Review by odinuge — APPROVED** [review]
/lgtm

**Comment by matthyx** [comment]
@tallclair that one was already assigned to you, thanks!

**Comment by matthyx** [comment]
@Random-Liu, @tallclair, @derekwaynecarr, @yujuhong, @ConnorDoyle this PR is really needed for 1.17, can you please make sure we don't miss code-freeze?

**Comment by Random-Liu** [comment]
/lgtm

We should get this change in after https://github.com/kubernetes/kubernetes/pull/84279

File: test/e2e_node/startup_probe_test.go [hunk 2]
```diff
@@ -58,15 +58,15 @@ var _ = framework.KubeDescribe("StartupProbe [Serial] [Disruptive] [NodeAlphaFea
 
 		/*
 			Release : v1.16
-			Testname: Pod liveness probe, using local file, delayed by startup probe
-			Description: A Pod is created with liveness probe that uses ‘exec’ command to cat the non-existent /tmp/health file. Liveness probe MUST NOT fail until startup probe expires.
+			Testname: Pod startup probe restart
+			Description: A Pod is created with a failing startup probe. The Pod MUST be killed and restarted incrementing restart count to 1, even if liveness would succeed.
 		*/
-		framework.ConformanceIt("should *not* be restarted with a exec \"cat /tmp/health\" because startup probe delays it [NodeConformance]", func() {
+		framework.ConformanceIt("should be restarted startup probe fails [NodeConformance]", func() {
```
Comment (liggitt) [inline test/e2e_node/startup_probe_test.go]: I'm very surprised these renames didn't require changing test/conformance/testdata/conformance.txt

@johnbelamaric, should this have touched that file?

Comment (liggitt) [reply test/e2e_node/startup_probe_test.go]: Hmm... looks like there are no files under test/e2e_node listed in conformance.txt. Are these actually considered conformance tests?

Comment (matthyx) [reply test/e2e_node/startup_probe_test.go]: Not sure... what's the requirement to consider them as conformance?

**Comment by johnbelamaric** [comment]
e2e_node does not run for conformance at all, so those are not really
conformance tests. In fact IIUC, tests in that directory do not run against
a full cluster but against some subset of components.

**Comment by matthyx** [comment]
Ok, understood.

As stated [here](https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/startup_probe_test.go#L48) these tests will be relocated once the feature gate is no longer needed.
I suppose we could call them conformance tests when this will have happened...

**Pull request merged**